### ISSUE TITLE
Can get_unchecked produce undefined behaviour?

### ISSUE THREAD

**Issue by axelkar** [cloudwego/sonic-rs#142]
If it doesn't, I think this should be added to the documentation. Maybe the unsafe keyword should be removed?

**Comment by liuq19** [comment]
yes, it will not validate the whole json and check utf8 here, and it may not return any error when using `get_unchecked` from an invalid json.

**Comment by axelkar** [comment]
What could happen if invalid JSON was provided? Is it only that LazyValue assumes it to be valid UTF-8? What about `get_from_str_unchecked`? Undefined behavior is documented [here](https://doc.rust-lang.org/reference/behavior-considered-undefined.html). I tried fuzzing `get_from_str_unchecked` after I made this issue, but nothing came up.

**Comment by liuq19** [comment]
The API may return unexpected results, such as not raising an error when invalid UTF-8 or invalid JSON. For example, `get_unchecked({"a": {invalidjson}, "b": 1}, &["b"])` will return the "1"  and not raise any errors

 It will not cause the undefined behavior (UB) defined in Rust.

**Comment by axelkar** [comment]
I think `unsafe` keyword should be removed for `get_from_str_unchecked` as it doesn't break safety guarantees anywhere. On the other hand, `get_unchecked` returns LazyValue, which assumes that it contains valid UTF-8.

**Comment by liuq19** [comment]
Not only does this involve invalid UTF-8, but calling `get_from_str_unchecked({"a": {invalidjson}, "b": 1}, &["b"])` will also not return an error, as it contains invalid JSON. This is because the `unchecked` APIs do not perform any validation on the JSON format.
